lua/telescope/builtin/files.lua

```diff
@@ -53,6 +53,26 @@ files.grep_string = function(opts)
   }):find()
 end
 
+-- NOTE(tj): Get clarification on comment and put this in the right place
+local function insert_search_dirs(find_command, search_dirs)
+  local command = find_command[1]
+  if command == 'fd' or command == 'fdfind' then
+    table.insert(find_command, '.')
+    for _,v in pairs(search_dirs) do
+      table.insert(find_command, v)
+    end
+  elseif command == 'rg' then
+    for _,v in pairs(search_dirs) do
+      table.insert(find_command, v)
+    end
+  elseif command == 'find' then
+    table.remove(find_command, 1)
```

Comment:
Here you need to expand dir paths, otherwise it results in error by users who add relative path to home dir

Reply:
Okay I did that, and found another issue. Even when an absolute path is given(relative to home), the search path tries to find it relative to the cwd after expanding. I'm looking into how to fix that now

Reply:
I'm having some trouble finding the source of the issue. At some point, the cwd is being prepended to the search directories, but I cannot figure out exactly where

Reply:
Is it because we've set cwd to some value perhaps?

```diff
   if command == 'fd' or command == 'fdfind' then
     table.insert(find_command, '.')
     for _,v in pairs(search_dirs) do
-      table.insert(find_command, v)
+      table.insert(find_command, vim.fn.expand(v))
     end
   elseif command == 'rg' then
     for _,v in pairs(search_dirs) do
-      table.insert(find_command, v)
+      table.insert(find_command, vim.fn.expand(v))
     end
   elseif command == 'find' then
     table.remove(find_command, 1)
     for _,v in pairs(search_dirs) do
-      table.insert(find_command, 2, v)
+      table.insert(find_command, 2, vim.fn.expand(v))
     end
   end
 end

```
Works well with me. Also it would be better if you moved the conditions inside the for loop so you don't end up with repeated code, don't you think?

Reply:
Sorry, was busy for a couple of days. I still cannot figure out where it's happening though. Do you think it's okay to ignore the absolute path case for now, and only allow relative paths to the cwd?
I also rewrote the loops and conditions as you suggested

Reply:
Nice looking forward to seeing the changes. For not using relative path case, I'm not sure why it didn't work for you, like what kinda of behavior are you getting?

The above diff works flawlessly for me.

```lua 
relativepath1 = "~/doc"
relativepath2 = "~/telescope"
builtin.fd({prompt_prefix = ">" , search_dirs =  [relativepath1 relativepath2] })

```

I'm sure I misunderstood what issue you are facing.

Reply:
I notice the following issues - 

1. If there are only absolute paths in the search dirs _and_ the cwd is the home directory, then the preview fails since it is getting the wrong path, but the correct file is opened
2. If there are a mixture of absolute and relative paths (or only absolute paths), and the cwd is the desired root of the relative paths (or anything other than the home directory), then bothe preview and file opening fails for the absolute paths

For example, running the following from my home directory
```lua
require("telescope.builtin").find_files {
		find_command={ 'rg', '--files'},
		search_dirs = {"~/.config/sxhkd", "~/.config/alacritty"},
		width = .25
}
```
results in the following error in the preview (but the correct files open when I press `<CR>` on them)-
![image](https://user-images.githubusercontent.com/49116134/100749204-5d9fc080-340a-11eb-85fc-30cc33c3d6a5.png)
If the cwd in neovim is not my home directory, for the same case I get the same errors in preview, and the wrong file opens
For the case you mentioned above, try running it from any directory other than your home directory. Does it work?

Reply:
As for relative paths with the correct cwd set it works perfectly fine. Since that is the only way I was using it, I did not notice this issue until you mentioned the expansion for supporting absolute dirs

Reply:
Oh you ware right, I see what you mean, I just test it out with preview enabled.